custom_components/myskoda/sensor.py

```diff
@@ -263,8 +263,10 @@ def native_value(self) -> int | float | None:  # noqa: D102
         if range := self.vehicle.driving_range:
             return range.total_range_in_km
 
-    def required_capabilities(self) -> list[CapabilityId]:
```

Comment:
Removing the `required_capabilities` will cause the entity to always be created. Wouldn't it be better here to return the relevant capability we have, such as 

```python
if self.vehicle.has_capability(CapabilityId.CHARGING):
  return [CapabilityId.CHARGING]
elif self.vehicle.has_capability(CapabilityId.XXX)
   etc
else
  return [CapabilityId.STATE]
   ```

Reply:
Can do. Starts to feel like it might make more sense to fully delegate the decision of 'being supported' to each entity class itself. E.g every `MySkodaEntity` would have a `is_supported` method (base class just returns `True`). Most would just inspect if one of more capabilities are present (or missing, in case of the forbidden_capabilities) but more advanced logic would then be possible. I also like how things are set up in the BMW integration: https://github.com/home-assistant/core/blob/c154ac26eb6287fa144cd48cbe8fd80bbe4fb09d/homeassistant/components/bmw_connected_drive/sensor.py#L58

Might take a look at that later. But out of scope here.

Reply:
I like that idea of the `is_supported` class, it should make this cleaner.

About the BMW integration: It has stuff I like, and stuff I like less. Let's pick the right things to re-use :-)

Reply:
As also just discussed in Discord, it would probably be better to override `is_supported()` in this class to something else than "just" the capabilities, but also take vehicle state into account.

Reply:
Updated